Extensions/messaging_tweaks.js

```diff
@@ -147,6 +152,32 @@ XKit.extensions.messaging_tweaks = new Object({
 				$(this).addClass("xkit-others_messaging_icon");
 				$(this).parents(".conversation-message").addClass("xkit-others_messaging_message");
 			}
+			if (XKit.extensions.messaging_tweaks.preferences.allow_emojis.value) {
+				var msg_div = $(this).parents(".conversation-message").find(".message-bubble .message");
+				
+				// Use regex to find emoji patterns in the chat message
+				// Find the string between the two ":" to find which image to use
+				// but only if there are emoji tags found
+				var emojis;
+				if (emojis = msg_div.text().match(/:(.*?):/gi)) {
+					for (var i = 0; i < emojis.length; i++) {
+						var emoji_tag = emojis[i].toLowerCase().substring(1).substring(0, emojis[i].length - 2);
+						var emoji_url = "http://www.emoji-cheat-sheet.com/graphics/emojis/" + emoji_tag + ".png";
+						if (emoji_tag === "tobdog") {
+							emoji_url = "http://vignette1.wikia.nocookie.net/steamtradingcards/images/2/21/Tobdog.png/revision/latest?cb=20151011230955";
+						}
+						var emoji = new Image();
+						// "꞉" is a unicode character (\uA789). Using that so having multiple emojis of the same type work properly
+						emoji.alt = "꞉" + emoji_tag + "꞉";
+						emoji.title = "꞉" + emoji_tag + "꞉";
+							emoji.height = "22";
+						emoji.onload = function(emoji_text) {;
```

Comment:
I started off writing a pointer on how to make this not trigger the "Creating functions in a loop" linting rule and ended up fixing the duplicate emoji problem without using the unicode colon:
```javascript
if (emojis) {
    var emoji_replacer = function(emoji_text, emoji_image) {
        msg_div.html(msg_div.html().replace(new RegExp(emoji_text, "g"), emoji_image.outerHTML));
    };

    // Remove duplicate emoji insertions to fix
    // double-replacement issues.
    var emoji_texts = {};

    for (var i = 0; i < emojis.length; i++) {
        emoji_texts[emojis[i]] = true;
    }

    for (var emoji_text in emoji_texts) {
        var emoji_tag = emoji_text.toLowerCase().substring(1).substring(0, emoji_text.length - 2);
        var emoji_url = "http://www.emoji-cheat-sheet.com/graphics/emojis/" + emoji_tag + ".png";
        if (emoji_tag === "tobdog") {
            emoji_url = "http://vignette1.wikia.nocookie.net/steamtradingcards/images/2/21/Tobdog.png/revision/latest?cb=20151011230955";
        }
        var emoji = new Image();
        emoji.alt = emoji_text;
        emoji.title = emoji_text;
        emoji.height = "22";
        emoji.onload = emoji_replacer.bind(null, emoji_text, emoji);
        emoji.src = emoji_url;
    }
}
```
Enjoy I guess?